spec/controllers/session_controller_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe SessionsController do
+  let!(:user) { User.create(name: "username", password: 'password', password_confirmation: 'password')}
+
+  describe 'GET #new' do
+    it "renders the :new view" do
+      get :new
+      expect(response).to render_template :new
+    end
+  end
+
+  describe 'POST #create' do
+
+    it "redirects to the user's :show page" do
+      post :create, user:{name: user.name, password: user.password}
+      expect(response).to redirect_to(user_path(user))
+    end
+
+    it "redirects to the sign in on error" do
+      post :create, user:{name: user.name, password: "not_password"}
+      expect(session[:user_id]).to eq(nil)
```

Comment:
This isn't what the description suggests it should be? This is a valid test but isn't testing what you claim it is.